javascript/ql/src/semmle/javascript/TypeAnnotations.qll

```diff
@@ -0,0 +1,104 @@
+/**
+ * Provides classes for reasoning about type annotations independently of dialect.
+ */
+
+import javascript 
+
+/**
+ * A type annotation, either in the form of a TypeScript type or a JSDoc comment.
+ */
+class TypeAnnotation extends @type_annotation {
```

Comment:
I suppose we could make this extend `Locatable`?

Reply:
Done, although the location of the JSDoc annotations seems to be `file://:0:0:0:0`. I'll try and see if there's a simple fix for that.

Reply:
> the location of the JSDoc annotations seems to be file://:0:0:0:0

That doesn't sound intentional. If you can fix it that would be great, but of course it's orthogonal to this PR.

Reply:
I think I fixed it. I'll just do it in the same PR.